pydis_site/apps/content/resources/guides/pydis-guides/contributing/bot.md

```diff
@@ -2,207 +2,643 @@
+                <li>Docker Compose. If you're using macOS and Windows, this already comes bundled with the previous installation. Otherwise, you can download it either from the <a href="https://docs.docker.com/compose/install">website</a>, or by running <code>pip install docker-compose</code>.</li>
+            </ul>
+            <p>If you get any Docker related errors, reference the <a href="../docker#possible-issues">Possible Issue</a> section of the Docker page.</p>
+        </div>
+    </div>
+</div>
+<br>
+
+Assuming you have Docker installed, enter the cloned repo in the command line and type `docker-compose up`.
+
+After pulling the images and building the containers, your bot will start. Enter your server and type `!help` (or whatever prefix you chose instead of `!`).
+
+Your bot is now running, but this method makes debugging with an IDE a fairly involved process. For additional running methods, continue reading the following sections.
+
+#### With the Bot Running Locally
+The advantage of this method is that you can run the bot's code in your preferred editor, with debugger and all, while keeping all the setup of the bot's various dependencies inside Docker.
+
+* [Prepare the hosts file on your machine](../hosts-file).
+* Append the following line to your `.env` file: `BOT_API_KEY=badbot13m0n8f570f942013fc818f234916ca531`.
+* In your `config.yml` file, set `urls.site` to `"pythondiscord.local:8000"`. If you wish to keep using `web:8000`, then [COMPOSE_PROJECT_NAME](../docker/#compose-project-names) has to be set to use this domain.
+
+You will need to start the services separately, but if you got the previous section with Docker working, that's pretty simple:
+
+* `docker-compose up web` to start the site container. This is required.
+* `docker-compose up snekbox` to start the snekbox container. You only need this if you're planning on working on the snekbox cog.
+* `docker-compose up redis` to start the Redis container. You only need this if you're not using fakeredis. For more info refer to [Working with Redis](#optional-working-with-redis).
+
+You can start several services together: `docker-compose up web snekbox redis`.
+
+##### Setting Up a Development Environment
+The bot's code is Python code like any other. To run it locally, you will need the right version of Python with the necessary packages installed:
+
+1. Make sure you have [Python 3.9](https://www.python.org/downloads/) installed. It helps if it is your system's default Python version.
+2. Install Poetry by following [these instructions](https://github.com/python-poetry/poetry#installation), or with `pip install poetry`. In case of the latter, make sure pip installs to the right Python version.
+3. [Install the dependencies](../installing-project-dependencies).
+
+Assuming you have at least the site running in Docker already, you can now start the bot locally through the command line, or through your preferred IDE. Notice that the bot is started as a module.
```

Comment:
This is a bold assumption. I feel like we should at least link to the site setup instructions at this point so they can go down that rabbit hole if they want.

Also "Notice that the bot is started as a module". I *think* I know what you mean, but I wouldn't expect someone to by default know what that means. I think having the command to start the bot here (above the collapsible section) would be helpful and then the collapsible section can go into further details. Or move the comment into the collapsible section.

Reply:
This section is directly until the part which mentions how to start the site container (line 521). Do you think it should be mentioned again?

Reply:
Ah, I missed it in the above paragraph. Maybe change it to: "The site has to have been started beforehand, see the above paragraph for the site contributing guide."

Also, for that above paragraph I would maybe change it to:
"You can run additional services on the host. This guide won't go over how to install and start the other dependencies on the host. If possible, prefer to start the services through Docker to replicate the production environment as much as possible.

The site, however, is a mandatory service for the bot. Refer to the [site contributing guide](linky link) to learn how to start it on the host, in which case you will need to change `urls.site` in `config.yml` to wherever the site is being hosted."

Reply:
I almost committed "linky link" <_<